spec.md

```diff
@@ -2221,108 +2298,65 @@ for referring to a declared circuit component, for statically and dynamically
 accessing a nested element within a component, for creating multiplexers, for
 performing primitive operations, and for reading a remote reference to a probe.
 
-## Unsigned Integers
+## Unsigned Hardware Integers
```

Comment:
"Consant Interger Expressions"?  In my mind `UIint...` is an expression (operator) which produces a value of unsigned integer type from a syntatic integer literal.

Reply:
This is the language that I was looking for! Thanks! These are expressions and that helps clarify everything.

Reply:
I talked with Adam a bit about this and he was more in favor of `UInt Literal` and `SInt Literal`. I started to write this and then switched back to `Constant Integer Literal`. The reason for this is that `UInt` and `SInt` are always referred to, in prose, as "unsigned integer type" or "signed integer type". I'd like to be able to keep the following distinctions to avoid collisions with "integer literal" which seems natural to mean `42` and not `UInt(42)`. The current language is as follows:

- "integer literal", e.g., 42
- "unsigned integer type", e.g., `UInt<8>`
- "constant integer expression", e.g., `UInt<8>(42)`

WDYT?